openmc/tallies.py

```diff
@@ -21,9 +22,58 @@ def reset_auto_tally_id():
 
 
 class Tally(object):
+    """A tally defined by a set of scores that are accumulated for a list of
+    nuclides given a set of filters.
+
+    Parameters
+    ----------
+    tally_id : int
```

Comment:
should there be any indication that the ``tally_id`` and ``name`` are optional and will be auto-generated if left undefined? This is question is not specific solely to the ``Tally`` class and is applicable to all other classes.